test/parallel/test-regress-GH-5727.js

```diff
@@ -17,14 +16,23 @@ net.Server().listen(0, function() {
 // The first argument is a configuration object
 assert.throws(() => {
   net.Server().listen({ port: invalidPort }, common.mustNotCall());
-}, errorMessage);
+}, common.expectsError({
+  code: 'ERR_SOCKET_BAD_PORT',
+  type: RangeError
+}));
 
 // The first argument is the port, no IP given.
 assert.throws(() => {
   net.Server().listen(invalidPort, common.mustNotCall());
-}, errorMessage);
+}, common.expectsError({
```

Comment:
It would be nice to change the pattern from `assert.throws(throwingFn, common.expectsError(obj))` to `common.expectsError(throwingFn, obj)`. But it should not block the PR.